honeybee/cli/edit.py

```diff
@@ -0,0 +1,292 @@
+            individual aperture centerlines.  If this number is larger than
+            the parent rectangle base, only one aperture will be produced.
+    """
+    try:
+        assert os.path.isfile(model_json), 'No JSON file found at {}.'.format(model_json)
+        # serialize the Model to Python
+        with open(model_json) as json_file:
+            data = json.load(json_file)
+        parsed_model = Model.from_dict(data)
+        # check the Model tolerance
+        assert parsed_model.tolerance != 0, \
+            'Model must have a non-zero tolerance to use windows-by-ratio-rect.'
+        tol = parsed_model.tolerance
+
+        # generate the windows for all walls of rooms
+        for room in model.rooms:
+            for face in room.faces:
+                if isinstance(face.boundary_condition, Outdoors) and \
+                        isinstance(face.type, Wall):
+                    face.apertures_by_ratio_rectangle(
+                        ratio, aperture_height, sill_height, horizontal_separation,
+                        vertical_separation, tol)
+
+        # write the new model out to the file or stdout
+        output_file.write(json.dumps(parsed_model.to_dict()))
+    except Exception as e:
+        _logger.exception('Model windows by ratio rect failed.\n{}'.format(e))
+        sys.exit(1)
+    else:
+        sys.exit(0)
+
+
+@edit.command('overhang')
+@click.argument('model-json', type=click.Path(
+    exists=True, file_okay=True, dir_okay=False, resolve_path=True))
+@click.argument('depth', type=float)
+@click.option('--angle', help='A number for the for an angle to rotate the overhang '
+              'in degrees. Positive numbers indicate a downward rotation while '
+              'negative numbers indicate an upward rotation.',
+              type=float, default=0, show_default=True)
+@click.option('--vertical-offset', help='An optional number for the vertical offset of '
+              'the overhang from the top of the window or face. Positive numbers move '
+              'up while negative mode down.', type=float, default=0, show_default=True)
+@click.option('--by-window/--by-wall', help='Flag to note whether the overhangs should '
+              'be generated per aperture or per wall.', default=True, show_default=True)
+@click.option('--outdoor/--indoor', help='Flag to note whether the overhangs should be '
+              'on the indoors like a light shelf.', default=True, show_default=True)
+@click.option('--output-file', help='Optional file to output the Model JSON string'
+              ' with overhangs. By default it will be printed out to stdout',
+              type=click.File('w'), default='-')
+def overhang(model_json, depth, angle, vertical_offset, by_window, outdoor, output_file):
+    """Add overhangs to all outdoor walls or windows in walls.\n
+    \n
+    Args:
+        model_json: Full path to a Model JSON file.\n
+        depth: A number for the overhang depth.
```

Comment:
What about adding multiple shades?

Reply:
I will add two separate commands for generating louvers by count and louvers by distance between. I just wanted to get your thoughts on this one first before adding them.